qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -218,13 +218,18 @@ def aux_operators(self,
         self._aux_op_nones = None
         if isinstance(aux_operators, list):
             self._aux_op_nones = [op is None for op in aux_operators]
-            zero_op = I.tensorpower(self.operator.num_qubits) * 0.0
-            converted = [op.to_opflow() if op else zero_op for op in aux_operators]
+            converted = [op.to_opflow() if isinstance(op, LegacyBaseOperator)
+                         else op for op in aux_operators]
             # For some reason Chemistry passes aux_ops with 0 qubits and paulis sometimes.
+            zero_op = I.tensorpower(self.operator.num_qubits) * 0.0
             converted = [zero_op if op == 0 else op for op in converted]
-            aux_operators = ListOp(converted)
```

Comment:
Also this conversion to `ListOp` should stay in place, otherwise the type of `aux_operators` is `list` and not `ListOp` (which actually is an Operator 🙂

Reply:
- That is right; I thought list is the intended type of ``aux_operators``, according to the documentation of the getters in [VQE](https://qiskit.org/documentation/stubs/qiskit.aqua.algorithms.VQE.aux_operators.html?highlight=vqe%20aux_operators) and [NumPyEigensolver](https://qiskit.org/documentation/stubs/qiskit.aqua.algorithms.NumPyEigensolver.aux_operators.html?highlight=aux_operators#qiskit.aqua.algorithms.NumPyEigensolver.aux_operators). The setter in NumPyEigensolver keeps the type of ``aux_operators`` as ``list``.
- Of course, since ``ListOp`` is iterable, this does not cause any issues. Semantically though, different elements of ``aux_operators`` could correspond to separate, independent observables. Additional functionality of ``ListOp``, such as combining all the operators with ``combo_fn``, wouldn't be desirable in that case.

Reply:
Each aux_operator was expected to be an independent observable, like a dipole moment, spin etc. They were supplied as a list. Internally how these are processed/computed is up to the algorithm as long as the external interface is the same. The fact that they might be bundled into a ListOp for convenience for the algorithm should be just an internal detail that is not exposed.

Reply:
@woodsp-ibm, I believe the current behavior is such that the getter can return ``ListOp``.